src/stash/src/lib.rs

```diff
@@ -623,7 +680,7 @@ where
             }
             collection.append_to_batch(&mut batch, &k, &v, 1);
         }
-        stash.append(once(batch))?;
+        stash.append(once(batch)).await?;
```

Comment:
What exactly does it mean if it is not cancel safe? I'm not sure I understand what that comment is requiring callers do or not do.

Reply:
The best documentation of cancel safety that I know of is buried in the `tokio::select` docs: https://docs.rs/tokio/latest/tokio/macro.select.html#cancellation-safety

That said, I'm not sure what makes it not cancel safe! (Though I'm not looking at the code right now, sorry!) What goes wrong if you drop the future? You just lose track of whether the transaction was committed or not, right? You shouldn't violate atomicity or anything, right?

Reply:
That's my understanding, yes. All operations in any impl are in transactions. If the caller crashes, it'll find out on restart where it left off, and it never ack'd to the user, so that's fine.

Reply:
@benesch I think my concern is that nothing about the stash api prevents implementors from performing 2 different transactions. Cancelling the future in between would be confusing (admittedly, we need to handle crashes so we aren't free to do what we want here). Another example is, after the `.commit().await`, if you do anything that `.await`'s, you can lose information on whether a transaction succeeded.

It sounds like we are actually claiming that the stash api assumes that the implementations are transactional; that is, they don't commit anything unless they returned `Poll::Ready`. If thats the case, then we we should document that understanding, as opposed to "these are not cancel safe"